app/web/src/components/ModelingView/TemplateSelectionModal.vue

```diff
@@ -169,7 +171,7 @@ const onCreateTemplate = () => {
     category: category.value,
   };
 
-  componentsStore.CREATE_TEMPLATE_FUNC_FROM_COMPONENTS(templateData);
+  componentsStore.CREATE_TEMPLATE_FUNC_FROM_COMPONENTS(templateData, router);
```

Comment:
is there a reason to pass the router, rather than import the router in component store?

Reply:
For some reason, we were not able to get the router imported successfully within the component store. It would only result in "undefined" no matter where we imported it within the store. Same issue within the toast itself. This workaround avoids that problem.

Reply:
useRouter does not work inside of the toast component, something about they way they are mounted

Reply:
also does not work inside stores!

Reply:
Ohhhhh, how interesting... because we definitely do router.push in view store for example?? Or am I losing my mind? Highly likely